src/library/dao/playlistdao.cpp

```diff
@@ -414,6 +414,37 @@ void PlaylistDAO::removeHiddenTracks(const int playlistId) {
     emit(changed(playlistId));
 }
 
+void PlaylistDAO::removeEmptyPlaylists(const int hidden) {
+    ScopedTransaction transaction(m_database);
+    // This query deletes all tracks marked as deleted and all
+    // phantom track_ids with no match in the library table
+    QString queryString = QString(
+        "SELECT id FROM Playlists "
+        "WHERE hidden=%1 AND "
+        "0=(SELECT COUNT(*) FROM PlaylistTracks  "
+        "   WHERE playlist_id = Playlists.id)")
+            .arg(QString::number(hidden));
+
+    QSqlQuery query(m_database);
+    if (!query.prepare(queryString)) {
+        LOG_FAILED_QUERY(query);
+        return;
+    }
+    query.setForwardOnly(true);
```

Comment:
You don't need this if you use our FwdSqlQuery. Usage examples can be found in CrateStorage.